silo/views.py

```diff
@@ -697,74 +701,96 @@ def showRead(request, id):
-        if response.status_code == 401:
-            logout(request)
-            redirect('/')
-
-        print(data)
-        """
-        excluded_fields = excluded_fields + ['username', 'password', 'file_data','autopush_frequency']
-        excluded_fields = excluded_fields + ['autopull_frequency', 'read_url']
+        excluded_fields = excluded_fields + ['username', 'password',
+                                             'file_data',
+                                             'autopush_frequency',
+                                             'autopull_frequency', 'read_url']
 
     if request.method == 'POST':
-        form = get_read_form(excluded_fields)(request.POST, request.FILES, instance=read_instance)
+        form = get_read_form(excluded_fields)(request.POST, request.FILES,
+                                              instance=read_instance)
         if form.is_valid():
             read = form.save(commit=False)
             if read.username and read.password:
-                basic_auth = base64.encodestring('%s:%s' % (read.username, read.password))[:-1]
+                basic_auth = base64.encodestring('%s:%s' % (read.username,
+                                                            read.password)
+                                                 )[:-1]
                 read.token = basic_auth
                 read.password = None
             if form.instance.autopull_frequency:
-                read.autopull_expiration = datetime.datetime.now() + datetime.timedelta(days=170)
+                read.autopull_expiration = datetime.datetime.now() + \
+                                           datetime.timedelta(days=170)
             if form.instance.autopush_frequency:
-                read.autopush_expiration = datetime.datetime.now() + datetime.timedelta(days=170)
+                read.autopush_expiration = datetime.datetime.now() + \
+                                           datetime.timedelta(days=170)
 
             read.save()
             if form.instance.type.read_type == "CSV":
                 return HttpResponseRedirect("/file/" + str(read.id) + "/")
             elif form.instance.type.read_type == "JSON":
-                return HttpResponseRedirect(reverse_lazy("getJSON")+ "?read_id=%s" % read.id)
+                return HttpResponseRedirect(reverse_lazy("getJSON") +
+                                            "?read_id=%s" % read.id)
             if form.instance.type.read_type == "OneDrive":
-                return HttpResponseRedirect("/import_onedrive/" + str(read.id) + "/")
-
-            if form.instance.autopull_frequency or form.instance.autopush_frequency:
-                messages.info(request, "Your table must have a unique column set for Autopull/Autopush to work.")
+                extra_data = {"token_type": "Bearer", "access_token":
+                    form.cleaned_data["onedrive_access_token"]}
+
+                social_auth = UserSocialAuth.objects.filter(user=request.user,
```

Comment:
It's better to use the `get_or_create` and then you update the `extra_data` field.

Reply:
I had some issues with `get_or_create` in this case, but i will check again.